interfaces/devdoc/arithmetic_requirements.md

```diff
@@ -0,0 +1,24 @@
+# arithmetic
+================
+
+## Overview
+
+The `arithmetic` module provides a platform/compiler-independent API for arithemtic operations on 128 bit numbers.
+
+-`umul64x64`: multiplies two uint64_t.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, uint64_t, umul64x64, uint64_t, left, uint64_t, right, uint64_t* out);
+```
+
+## umul64x64
+
+```c
+MOCKABLE_FUNCTION(, uint64_t, umul64x64, uint64_t, left, uint64_t, right, uint64_t* high);
+```
+
+`umul64x64` multiplies `left` with `right`, outputs in `high` the most significant 64 bits of the multiplication and returns the least significant 64bits of the multiplication. 
```

Comment:
> outputs in `high` the most significant 64 bits of the multiplication and returns the least significant 64bits of the multiplication.  [](http://example.com/codeflow?start=43&length=134)

I kind of feel like it's nice to either return both (struct) or put both as out parameters. As it is, it feels kind of mixed/inconsistent.

Reply:
Sorry, this was an old file :(. Updated with the intended content.